documentation/release_notes.markdown

```diff
@@ -4,116 +4,104 @@ title: "Puppet Server: Release Notes"
 
-### New feature: Faster service restarts with HUP signals
+Puppet Server 2.4.0 respects the new [`always_retry_plugins` setting introduced in Puppet 4.5](https://docs.puppet.com/puppet/latest/reference/configuration.html#alwaysretryplugins), which determines how Puppet caches attempts to load Puppet resource types and features. However, Server changes this setting's value from its default (true) to false, in order to take advantage of additional caching for failures when loading types.
 
-Puppet Server 2.3.0 and newer support being restarted by sending a hangup signal, also known as [HUP or SIGHUP](./restarting.markdown), to the running Puppet Server process. You can send this signal to the Puppet Server process using the standard `kill` command. The HUP signal stops Puppet Server and reloads it gracefully, without terminating the JVM process. This is generally much faster than completely stopping and restarting the process, and allows you to quickly load changes to your Puppet Server master, including certain configuration changes.
+The `always_retry_plugins` setting also replaces the [`always_cache_features` setting](https://docs.puppet.com/puppet/4.5/reference/configuration.html#alwayscachefeatures), which is now deprecated. If you set `always_cache_features` to true in previous versions of Puppet Server, set `always_retry_plugins` to false.
 
-* [SERVER-96](https://tickets.puppet.com/browse/SERVER-96)
+-   [PUP-5482](https://tickets.puppetlabs.com/browse/PUP-5482)
 
-### Bug fix: Puppet Server correctly parses complex script arguments
+### New feature: Expanded logging for certificate autosigning attempts
 
-In versions 1.x and 2.2.x, Puppet Server would incorrectly parse commands executed by Puppet code that had complex string interpolation. For example, calls to the `generate()` function --- such as `generate('/bin/sh', '-c', "/usr/bin/python -c 'print \"foo\"'")` --- would spawn a Python REPL and consume a JRuby instance without returning anything. Puppet Server 2.3.0 fixes this issue.
+Starting with version 2.4.0, Puppet Server [logs](./configuration.markdown#logging) message and warnings when an autosign command generates STDERR output or returns a non-zero exit code. Server 2.4.0 also logs autosigning attempts at the INFO level, rather than DEBUG, to help make autosigning issues easier to diagnose without changing Server's logging level.
 
-* [SERVER-1160](https://tickets.puppet.com/browse/SERVER-1160)
+-   [SERVER-1187](https://tickets.puppetlabs.com/browse/SERVER-1187)
 
-### Known issues
+### Bug fix: Closed memory leak when restarting Server via SIGHUP
 
-#### Modifications to `bootstrap.cfg` might cause problems during upgrades to 2.3.0
+The versions of Trapperkeeper included with Puppet Server 2.3.x leaked a small amount of memory when [restarting Server with a HUP signal](./restarting.markdown). Trapperkeeper 1.4.0, which is included with Puppet Server 2.4.0, resolves this issue.
```

Comment:
Yeah, I'd probably just say something like "The versions of the Trapperkeeper components include with...", and then remove the sentence at the end about TK1.4.0.  Presumably if anyone really cares about which component had the issue or what version of the component fixed the issue, they can just click through to the Jira?